### ISSUE TITLE
camelcase rule should ignore global variables by default

### ISSUE THREAD

**Issue by hrax** [eslint/eslint#11716]
**What rule do you want to change?**
camelcase

**Does this change cause the rule to produce more or fewer warnings?**
Change should produce fewer warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**
Change should be implemented as a new default behavior.

**Please provide some example code that this change will affect:**

```js
/* globals g_form */
g_form.setValue("field", "value");
```

**What does the rule currently do for this code?**
Produces an error "Identifier 'g_form' is not in camel case." unless the variable is set as exception in camelcase rule allow property.

**What will the rule do after it's changed?**
In case the defined global variable is not present for the script, setting the variable as an exception in the camelcase rule would allow the developer to set said variable in the script and therefore avoid checks for camelcase. This is even more potentially harmful if you set up these variables as a regular expression.

**Are you willing to submit a pull request to implement this change?**
Not at the moment.

**Comment by platinumazure** [comment]
Hi @hrax, thanks for the issue.

I could support a new option for this, but not a default behavior change.

**Comment by hrax** [comment]
Hi @platinumazure, thank you for the reply. 
New option sounds great as well.
